include/onnxruntime/core/common/eigen_common_wrapper.h

```diff
@@ -60,8 +60,14 @@
 #pragma warning(disable : 4245)
 #pragma warning(disable : 4127)
 #endif
-
+#if defined(__GNUC__) and __cplusplus >= 202002L
```

Comment:
We can remove the "and __cplusplus >= 202002L" part as long as this suppression only applies to external header files.